### DISCUSSION TITLE
Question about tag page links to posts

### DISCUSSION THREAD

**Discussion by RCHG** [getpelican/pelican#3057] [Q&A]
Hello!

Hoping you are doing great. At this moment, I have a new blog with new template working well locally except for one aspect: in the page auto-generated for each tag or category the links to the post are not working.

In my **index.html** everthing works well the the links to the articles are ok, for example,

`http://127.0.0.1:8000/articles/tutorial/2022-01/tutorialEnergyBudget.html`

however in the list of articles per category or tags it points to a wrong place (tag is added to the path). For example this article has a tag named energy, and in the auto-generated page for that tag is:

`http://127.0.0.1:8000/tag/articles/tutorial/2022-01/tutorialEnergyBudget.html  `

and in the category the same is happening:

`http://127.0.0.1:8000/category/articles/tutorial/2022-01/tutorialEnergyBudget.html`

if I print the value of the variable **article.url** it is ok, so I can't not figure out why Pelican is adding *tag* or *category* to the path.

Note: in my pelicanconf.py it is added:

```
ARTICLE_URL     = 'articles/{category}/{date:%Y}-{date:%m}/{slug}.html'
ARTICLE_SAVE_AS = 'articles/{category}/{date:%Y}-{date:%m}/{slug}.html'
PAGES_URL       = 'pages/{slug}.html'
PAGES_SAVE_AS   = 'pages/{slug}.html'
```

Thank you!

**Comment by avaris** [comment]
Can you show your `tag.html` and `category.html` templates?

**Comment by RCHG** [comment]
Thank you. 

Here is the category.html and tag.html

```
{% extends 'base.html' %}
{% block content %}

    <h1>List of Articles with {{ category }}</h1>
    <table id="tbidx">
          <colgroup>
            <col span="1" style="width: 22%;">
            <col span="1" style="width: 78%;">
          </colgroup>
          <thead></thead>
    <tbody>
        {% for article in dates  %}
        <tr>
            <td id="tbidxdate" >{{ article.locale_date }}</td>
            <td id="tbidxtitle" > <a href="{{ article.url }}" title="{{ article.title }}">{{ article.title }}</a>{% for articletag in article.tags %}<a href="{{ SITEURL }}/{{ articletag.url }}"><span class="tag">{{ articletag }}</span></a>{% endfor %}
            </td>
        </tr>
        {% endfor %}
    </tbody>
    </table>
{% endblock content %}
```

```
{% extends 'base.html' %}
{% block content %}

    <br>
    <h1>List of Articles tagged with {{ tag }}</h1>
    <table id="tbidx">
          <colgroup>
            <col span="1" style="width: 22%;">
            <col span="1" style="width: 78%;">
          </colgroup>
          <thead></thead>
    <tbody>
        {% for article in dates  %}
        <tr>
            <td id="tbidxdate" >{{ article.locale_date }}</td>
            <td id="tbidxtitle" > <a href="{{ article.url }}" title="{{ article.title }}">{{ article.title }}</a>{% for articletag in article.tags %}<a href="{{ SITEURL }}/{{ articletag.url }}"><span class="tag">{{ articletag }}</span></a>{% endfor %}
            </td>
        </tr>
        {% endfor %}
    </tbody>
    </table>
    
{% endblock content %}
```

**Reply by avaris** [reply]
You need to have `SITEURL` in the `href` like you do in the other places (`href="{{ SITEURL }}/{{ articletag.url }}"`):

```jinja
<td id="tbidxtitle" > <a href="{{ SITEURL }}/{{ article.url }}" title="{{ article.title }}"> ...
```

**Reply by RCHG** [reply]
Thank you! I think it works well now!
